openedx_learning/apps/authoring/containers/api.py

```diff
@@ -0,0 +1,334 @@
+        # 1. The changes provoking a new version are the addition, removal of members or reordering.
+        # 2. Published and draft versions are always the latest for all members.
+        # 3. When creating a new version, a new user-defined entity list is created to preserve the latest state as the previous user-defined list.
+        # TODO: instead consider copying the previous user-defined list as the frozen list, and add/remove to the previous user-defined list.
+        # If it's a reordering, the previous user-defined list is copied as the frozen and a new user-defined list is created with the new order.
+        new_user_defined_list = create_entity_list_with_rows(
+            entity_pks=publishable_entities_pk,
+            draft_version_pks=[None] * len(publishable_entities_pk),
+            published_version_pks=[None] * len(publishable_entities_pk),
+        )
+        container_version = ContainerEntityVersion.objects.create(
+            publishable_entity_version=publishable_entity_version,
+            container_id=container_pk,
+            defined_list=new_user_defined_list,
+            initial_list=last_version.initial_list,
+            frozen_list=last_version.defined_list,
+        )
+    return container_version
+
+
+def create_container_and_version(
+    learning_package_id: int,
+    key: str,
+    created: datetime,
+    created_by: int | None,
+    title: str,
+    publishable_entities_pk: list[int],
+) -> ContainerEntityVersion:
+    """
+    Create a new container and its first version.
+
+    Args:
+        learning_package_id: The ID of the learning package that contains the container.
+        key: The key of the container.
+        created: The date and time the container was created.
+        created_by: The ID of the user who created the container.
+        version_num: The version number of the container.
+        title: The title of the container.
+        members_pk: The IDs of the members of the container.
+
+    Returns:
+        The newly created container version.
+    """
+    with atomic():
+        container = create_container(learning_package_id, key, created, created_by)
+        container_version = create_container_version(
+            container_pk=container.publishable_entity.pk,
+            version_num=1,
+            title=title,
+            publishable_entities_pk=publishable_entities_pk,
+            entity=container.publishable_entity,
+            created=created,
+            created_by=created_by,
+        )
+    return (container, container_version)
+
+
+def get_container(pk: int) -> ContainerEntity:
+    """
+    Get a container by its primary key.
+
+    Args:
+        pk: The primary key of the container.
+
+    Returns:
+        The container with the given primary key.
+
+    TODO: should this use with_publishing_relations as in components?
+    """
+    return ContainerEntity.objects.get(pk=pk)
+
+
+def get_defined_list_for_container_version(
```

Comment:
I wonder whether we should provide this kind of API method or whether users of the containers API should be unaware of EntityListRows' existence.